scripts/us_census/acs5yr/subject_tables/common/data_loader.py

```diff
@@ -0,0 +1,467 @@
+                obs_df['Unit'] = unit
+
+                # add scaling factor to the csv
+                try:
+                    scalingFactor = column_map[column]['scalingFactor']
+                    del column_map[column]['scalingFactor']
+                except KeyError:
+                    unit = np.nan
+                    scalingFactor = np.nan
+
+                # if StatVar not in mcf_dict, add dcid
+                dcid = column_map[column]['Node']
+
+                if dcid not in self.mcf_dict:
+                    ## key --> node dcid
+                    self.mcf_dict[dcid] = {}
+                    ## add pvs to dict
+                    for key, value in column_map[column].items():
+                        if key != 'Node':
+                            self.mcf_dict[dcid][key] = value
+
+                obs_df['ScalingFactor'] = scalingFactor
+                obs_df['Year'] = year
+                obs_df['Column'] = column
+
+                # Replace empty places (unresolved geoIds) as null values
+                obs_df['Place'].replace('', np.nan, inplace=True)
+
+                # Drop rows with observations for empty (null) values
+                obs_df.dropna(subset=['Place', 'Quantity'],
+                              axis=0,
+                              inplace=True)
+
+                # Write the processed observations to the clean_csv
+                if self.year_count == 0:
+                    obs_df.to_csv(csv_file, header=True, index=False, mode='w')
+                else:
+                    obs_df.to_csv(csv_file, header=False, index=False, mode='a')
+                self.year_count += 1
+
+                # update stats for the year:
+                self.counter_dict[year]["number of unique geos"] = len(
+                    obs_df['Place'].unique())
+                self.counter_dict[year][
+                    "number of observations"] += obs_df.shape[0]
+                self.counter_dict[year][
+                    "number of unique StatVars with observations"] += len(
+                        obs_df['StatVar'].unique())
+                self.counter_dict[year]["number of StatVars in mcf_dict"] = len(
+                    list(self.mcf_dict.keys()))
+        csv_file.close()
+        print(
+            f"""Completed with { self.counter_dict[year]['number of observations'] }
+            observation for { self.counter_dict[year]['number of unique StatVars with observations'] }
+            StatVars at { self.counter_dict[year]['number of unique geos'] } places.
+            """,
+            flush=True)
+
+    def _generate_mcf_from_column_map(self):
+        """
+        uses column_map to generate a mcf file with all the stat_vars nodes that
+        are created.
+        """
+
+        #get cleaned column map
+        final_mcf = ""
+
+        for column, stat_var in self.mcf_dict.items():
+            col_stat_var = []
+            dcid = 'Node: ' + column + '\n'
```

Comment:
add this to col_stat_var itself instead of an extra var for dcid?

can replace 351,352 with
 final_mcf = final_mcf + '\n'.join(col_stat_var) + "\n\n"